pkg/controllers/resources/persistentvolumes/syncer.go

```diff
@@ -130,13 +130,16 @@ func (s *persistentVolumeSyncer) Sync(ctx *synccontext.SyncContext, pObj client.
 	// check if objects are getting deleted
 	if vObj.GetDeletionTimestamp() != nil {
 		if pObj.GetDeletionTimestamp() == nil {
-			ctx.Log.Infof("delete physical persistent volume %s, because virtual persistent volume is terminating", pObj.GetName())
-			err := ctx.PhysicalClient.Delete(ctx.Context, pObj)
-			if err != nil {
-				return ctrl.Result{}, err
+			// check if the PV is dynamically provisioned and the reclaim policy is Delete
+			if !(vPersistentVolume.Spec.ClaimRef != nil && vPersistentVolume.Spec.PersistentVolumeReclaimPolicy == corev1.PersistentVolumeReclaimDelete) {
```

Comment:
Could you please explain in a bit more detail why this condition is needed?
Wouldn't this mean that PVs with other reclaim policy and without a claimRef would never get deleted? :thinking: 

Answers, and the PR, are much appreciated! Thank you.

Reply:
The NOT condition in this check will prevent the deletion of a PV which is dynamically provisioned and the reclaim policy is Delete. For all other conditions, it will go ahead and delete the physical PV.

Reply:
Oh damn, I need to clean my glasses :joy:. Sorry, I didn't notice the `!` right after `if` :man_facepalming: 

Okay, just so I understand completely, this will delay the deletion of the dynamically provisioned PVs with the reclaim policy is Delete. Once the vPV is deleted, the `syncUp` will be called and pPV will be deleted at that point?

Reply:
Ok, now it finally clicked. Thank you for the explanation and for your patience!! :raised_hands: